### DISCUSSION TITLE
CPU locked at lowest frequency (On behalf of other users who met this issue)

### DISCUSSION THREAD

**Discussion by sklynic** [seerge/g-helper#207] [General]
In short: I recommended this repo to lots of TUF users and some of them reported this weird problem. As they did't give me much details and logs, I think I can't post it to issues, so it's here. (Also there 's no post about this weird problem)

Description:
CPU frequency locked at 0.32/0.30Ghz after using g-helper on TUF laptop (happen on 2022 models using intel CPU)

How to reproduce:
Simply run g-helper.exe.  (ArmouryCrate was fully uninstalled)

OSes:
Both windows10/Windows11 have this behavior.

laptoop models:
See above, only 2022 models of TUF laptop using intel CPU (fow now I haven't receive any report happening on AMD model)

Additional details (very limited):
1.Some reported adjusting fan curve by 3rd-party software would trigger this issues (one mentioned atrofac).
2.Set higher CPU SPPT may reduce the chance of this problem (one reported set it above 65W)

As in this situation, it's hard to trace the bug and make fix (even gather more information is tough), so I opened this discussion.

**Comment by seerge** [comment]
It would be helpful to get app logs at least, cause i can’t debug much remotely:))

Does it happen if they apply power limits? or without them ? and what modes?

**Reply by seerge** [reply]
Update: there is a topic on tuf reddit 
https://www.reddit.com/r/Asustuf/comments/12esesy/comment/jfgoxsf/

It could be that some tuf models (probably, as you say intel-based from 2022) require either both PPTs & Fan curve to be applied or none. After all in armoury you can't set only fan curve without PPTs.

**Reply by sklynic** [reply]
> It would be helpful to get app logs at least, cause i can’t debug much remotely:))

I should add a how-to about getting log everytime I share this repo (since I haven't got any log from them so far)...don't mind

> Does it happen if they apply power limits? or without them ? and what modes?

for power limits: One guy mentioned about CPU PPT (see above), waitng for more report
modes: all three modes (silent, balanced, turbo) have this problem, changing between them don't slove it

> Update: there is a topic on tuf reddit https://www.reddit.com/r/Asustuf/comments/12esesy/comment/jfgoxsf/

Mentioned BD prochot...
Searched on internet and found tons of cases that cpu was locked at low frequency (including old tuf laptops). Is that the problem? 
But they say can't disable it on 12th gen intel for now...I'll check if other software can adjust this setting

> It could be that some tuf models (probably, as you say intel-based from 2022) require either both PPTs & Fan curve to be applied or none. After all in armoury you can't set only fan curve without PPTs.

If that's the reason, maybe we could try out bind it when set the two settings?

**Reply by seerge** [reply]
@sklynic i don't think all that "BD prochot" is related :) 

What happens to the guy on reddit - is that power for CPU is set to something ridiculously low (like 5w). Most probably due to a bugged bios implementation on that serie of laptops as it's not g-helper setting that for sure (i checked his logs). My guess - their bios requires both fan curve + ppts, and if only fan curve is set - then bios probably "defaults" to 5w ppt.

Just in case - you can manually set 5w to power in balanced mode - and you will indeed get something like 0.3ghz in CPU (but that's expected)

But I need more reproduction scenarios before trying to "fix" asus bios from my app 😄

**Comment by sklynic** [comment]
More people reported set custom fan curve will let the cpu frequency limited to 0.3Ghz or so. Sorry there's still no log I can offer...:(

> Just in case - you can manually set 5w to power in balanced mode - and you will indeed get something like 0.3ghz in CPU (but that's expected)

Personally, some time I does set cpu power to 5w using ryzen-controller when using battery to reach about 7w discharge rate (if not, then it's something in neighborhood of 12w), however it never drop to 0.30Ghz on my laptop, maybe that's a difference between Intel and AMD?
I'll try using g-helper to set 5w power and to see if there's any difference

> But I need more reproduction scenarios before trying to "fix" asus bios from my app 😄

You don't need to "fix" it. 
Thats asus's problem, nobody know what shoutcut they take in designing their software or bios or drivers. 
If the situation is explained clearly, then people will understand and accept it

**Reply by seerge** [reply]
@sklynic i have amd, and in balanced i managed to lower it to 0.3 :) (obviously it's not practical to do so)

As for fix - can you ask for model names at least? I know that FX507 seem to misbehave.

And they can try this build :
[GHelper.zip](https://github.com/seerge/g-helper/files/11185605/GHelper.zip)

It will "set" 80w PPT with custom fan curve (if no other PPTs defined by user)

**Reply by sklynic** [reply]
> As for fix - can you ask for model names at least? I know that FX507 seem to misbehave.

models:
FX517ZC(which cpu is i5 12650h),FX507ZC,FX507ZE
(If I'm correct, all the intel-based 2022 tuf models asus have made are fx507,fx517,fx707?)

> And they can try this build : [GHelper.zip](https://github.com/seerge/g-helper/files/11185605/GHelper.zip)
> 
> It will "set" 80w PPT with custom fan curve (if no other PPTs defined by user)

will let them try out and gather app log

**Reply by seerge** [reply]
@sklynic i have added fx507 + fx517 to the list where app will try to "tackle" bugged bios and pushed it as https://github.com/seerge/g-helper/releases/tag/v0.49 (so your friends can just try that)

[GHelper.zip](https://github.com/seerge/g-helper/files/11186266/GHelper.zip)

**Comment by sklynic** [comment]
Seems to be asus side's bug. Cpu frequency behave normal but gpu fan is 0% (according to this user, ArmouryCrate also have this problem when gpu mode is hybrid and using manual mode to adjust settings).

If choose Turbo cpu frequency still locks down
(edit: he said in armoury crate, after switching to turbo cpu frequency is normal)

![IMG_20230410_071923](https://user-images.githubusercontent.com/39230198/230801422-49ffa04d-6331-4241-998f-48e0786bad58.jpg)

log:
https://pastebin.ubuntu.com/p/RJBP6VYscX/

If that's asus side's bug, then I think there's only a little things we can do.

**Reply by seerge** [reply]
@sklynic i don't think it's related. On a screenshot i see gpu temp 47C and GPU chart starts somewhere from 50C.
As for power limits - reddit user said that with 0.49 it's ok for him now, so looks like my solution worked (i.e. applying manual limits every time  custom fans are set)

You can ask your friend to re-download 0.49 (it was updated with one more tweak for power). As it turned out there is actually 2 separate endpoints that set power for short and long times - and now app "fixes" asus bios with setting both values

**Reply by sklynic** [reply]
> On a screenshot i see gpu temp 47C and GPU chart starts somewhere from 50C.

sorry  i put a wrong image:(
![d320d673ec7ed909388fb90f15c56c88](https://user-images.githubusercontent.com/39230198/230870300-2357db32-64ba-44b6-afde-bd57f806050c.jpeg)

> You can ask your friend to re-download 0.49 (it was updated with one more tweak for power). As it turned out there is actually 2 separate endpoints that set power for short and long times - and now app "fixes" asus bios with setting both values

will let him try out as soon as possible

**Reply by seerge** [reply]
@sklynic also such fan curves can make bios go nuts i think (i.e. when everything is at max), i would try something more realistic 😄 

And again - it's not really related to main issue : bios setting incorrect power when custom ppts were off / custom fans on

P.S. I see that they has a whole zoo of apps , including atrofac (that also can set fan curve)

**Reply by sklynic** [reply]
> P.S. I see that they has a whole zoo of apps , including atrofac (that also can set fan curve)

I mentioned the atrofac at first. That's from this guy's report.
So this issue could be seen as solved? Then I'll close this discussion when get his feedback that his problem is solved.

**Reply by seerge** [reply]
@sklynic well for "fans" - my advice would be not to max them out all time (as it's also bad for fans physically in a long term)

As for main issue - i believe so. Just make sure your friends re-download latest 0.49 :)

**Comment by sklynic** [comment]
CPU frequency problem is solved and I'm closing.

But a off topic problem: his gpu fan still don't work, same as manual in armoury (the first fan curve 1.is above Turbo's default fan profie, maybe still too much high? 2.30C's rpm is bigger than 40/50C, may get bios "confused"?)
first image is on balanced+ custom fan curve, second one is on turbo's default fan curve
![IMG_20230410_230628](https://user-images.githubusercontent.com/39230198/230930550-1d9558ab-ccff-43b2-9700-c6c64a0844bc.jpg)
![IMG_20230410_232641](https://user-images.githubusercontent.com/39230198/230932599-2901e2d1-a497-429b-9526-ed29db5ce009.jpg)

Another thing off topic but interesting:
https://gitlab.com/asus-linux/asusctl/-/issues/341
tuf's bios is notorious...ah, well...(would you mind me mention this discussion at there?)

**Reply by seerge** [reply]
@sklynic so when they set custom curve in armoury itself, it also doesn't work ?

**Reply by sklynic** [reply]
Yes. Choose Manual in armoury and set fan curve don't work (he mentioned that's on recent armoury version)

**Reply by seerge** [reply]
@sklynic well, if it's not working even in own asus app, then it's probably bugged so much :) i'm not sure if there is anything i can do (without access to device especially)
